orbit/pkg/table/tcc_access/tcc_access.go

```diff
@@ -0,0 +1,179 @@
+//go:build darwin
+// +build darwin
+
+package tcc_access
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+var (
+	testContext   = false
+	tccPathPrefix = ""
+	tccPathSuffix = "/Library/Application Support/com.apple.TCC/TCC.db"
+	dbQuery       = "SELECT service, client, client_type, auth_value, auth_reason, last_modified, policy_id, indirect_object_identifier, indirect_object_identifier_type FROM access;"
+	sqlite3Path   = "/usr/bin/sqlite3"
+	dbColNames    = []string{"service", "client", "client_type", "auth_value", "auth_reason", "last_modified", "policy_id", "indirect_object_identifier", "indirect_object_identifier_type"}
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		// added here
+		table.TextColumn("source"),
+		table.IntegerColumn("uid"),
+		// derived from a TCC.db
+		table.TextColumn("service"),
+		table.TextColumn("client"),
+		table.IntegerColumn("client_type"),
+		table.IntegerColumn("auth_value"),
+		table.IntegerColumn("auth_reason"),
+		table.BigIntColumn("last_modified"),
+		table.IntegerColumn("policy_id"),
+		table.TextColumn("indirect_object_identifier"),
+		table.IntegerColumn("indirect_object_identifier_type"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+// Constraints for generating can be retrieved from the queryContext.
+
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
```

Comment:
We should load the `"uid"` constraint of the query and only process the db files of the requested users. (E.g. if I do `SELECT * FROM tcc_access WHERE uid = 500;` then this table will (unnecessarily) load everything and then osquery will filter out the results instead of loading the TCC db of user with uid=500).

Reply:
Okay, implemented that filtering, including for the system db, and for all of `>`, `<`, `>=`, `<=`, and `=`. Here are some checks - in green is the query sent from Fleet, and the logs are from when Generate is actually querying a table:

<img width="1419" alt="Screenshot 2024-06-03 at 6 15 01 PM" src="https://github.com/fleetdm/fleet/assets/61553566/8dac8a96-b9ec-4cd7-af3d-a82a6c01db42">
<img width="1419" alt="Screenshot 2024-06-03 at 6 17 54 PM" src="https://github.com/fleetdm/fleet/assets/61553566/3fd7f050-fb41-47d9-b07f-b795d1a5a086">
<img width="1419" alt="Screenshot 2024-06-03 at 6 20 50 PM" src="https://github.com/fleetdm/fleet/assets/61553566/2d7d261b-cd11-4586-9a56-ef1d2960a327">
<img width="1419" alt="Screenshot 2024-06-03 at 6 19 31 PM" src="https://github.com/fleetdm/fleet/assets/61553566/ae9497fc-e9a5-49af-8c13-45b8c2da4737">